### ISSUE TITLE
python ints becomes double in matlab land

### ISSUE THREAD

**Issue by sdvillal** [arokem/python-matlab-bridge#165]
Just food for thought. The following little snippet shows the (potential) problem:

```python
from pymatbridge import Octave
o = Octave()
o.start()
o.set_variable('x', 12)
o.run_code('class(x)')
```
Being more catholic than the pope, class(x) in matlab land should probably be "int{32|64}", however it is "double". In theory and in my experience casting all numbers to floats in matlab land is a good default, that matches better what matlab expect (the literal "2" in matlab is a double, so it is str2num('2')). But perhaps there should be a way to move ints as ints, just in case someone ever find code that does expect ints (I have seen it, but do not ask me where). I think [oct2py approach](https://github.com/blink1073/oct2py/blob/812a1a43184c23a54c783817b67eaadd838e80e1/oct2py/core.py#L64-L65) is the way to go. 

An alternative is always to set the variable, run *'x=int64(x)'* and then run the call. But that is not as convenient.

**Comment by blink1073** [comment]
Hi @sdvillal, we could expose this as a setting (which is what Oct2Py [does](https://github.com/blink1073/oct2py/blob/master/oct2py/core.py#L64)).

**Comment by blink1073** [comment]
Ha, I just saw you pointed to the same line of code :smile:.

**Comment by sdvillal** [comment]
Yeah, I have a library that intercheangeable can use pieces of oct2py  and pieces of python-matlab-bridge for (data-transfer / running code) so I know both. And have been bridging matlab into other languages for more time than it is healthy.

After poking around a bit the code, I guess  implementing the int->int stuff can be involved. Maybe a parameter controlling the non-default behavior should be added to all functions and somehow routed into the request, then tweak json-decoding in matlab/octave to take it into account?

This is how I made my more catholic than the pope tests pass:
```python
def _put_hook(self, varnames, values, eng, int2float):
        for name, value in zip(varnames, values):
            eng.session().set_variable(name, value)
            if not int2float and isinstance(value, int):
                eng.run_command('%s = int64(%s)' % (name, name))  # FIXME: or int32, if we are in a 32 bit system
```

**Comment by sdvillal** [comment]
Just as a note, I still like more oct2py data transfer mechanism, specially if one uses ramdisks. Do not completelly abandon it ;-)
